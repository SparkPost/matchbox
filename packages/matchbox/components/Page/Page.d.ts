export default Page;
declare function Page(props: any): any;
declare namespace Page {
    const displayName: string;
    namespace propTypes {
        const title: any;
        const subtitle: any;
        const primaryAction: any;
        const primaryArea: any;
        const secondaryActions: any;
        const breadcrumbAction: any;
        const empty: {
            (props: any, propName: any, componentName: any, ...rest: any[]): any;
            isRequired: any;
        };
        const children: any;
    }
}
