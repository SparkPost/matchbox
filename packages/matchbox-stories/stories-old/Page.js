import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { action } from '@storybook/addon-actions';
import { StoryContainer } from './helpers';

import { Page, Panel } from '../src';

export default storiesOf('Page', module)
  .addDecorator((getStory) => (
    <StoryContainer>{ getStory() }</StoryContainer>
  ))

  .add('with all props', withInfo()(() => {
    const primaryAction = {
      content: 'Publish',
      onClick: action('Publish Clicked'),
      color: 'orange'
    };

    const secondaryActions = [
      {
        content: 'Save',
        onClick: action('Save Clicked')
      },
      {
        content: 'View Draft',
        onClick: action('Draft Clicked')
      },
      {
        content: 'Preview & Send',
        onClick: action('Preview Clicked')
      }
    ];
    const breadcrumbAction = {
      content: 'Templates',
      onClick: action('Templates Clicked')
    }
    return (
      <Page
        primaryAction={primaryAction}
        secondaryActions={secondaryActions}
        breadcrumbAction={breadcrumbAction}
        title='Template #3' >
        <Panel sectioned>Content</Panel>
      </Page>
    )
  }))

  .add('with an empty state', withInfo()(() => {
    const primaryAction = {
        content: 'Create',
        onClick: action('Create Clicked'),
        color: 'orange'
      };
    return (
      <Page
        empty={{
          show: true,
          title: 'Empty State Title',
          content: <p>Empty State Content</p>
        }}
        primaryAction={primaryAction}
        title='Template #3'
      />
    )
  }));
