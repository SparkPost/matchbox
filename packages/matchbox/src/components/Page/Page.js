import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { buttonFrom } from '../Button';
import { ChevronLeft } from '@sparkpost/matchbox-icons';
import { EmptyState } from '../EmptyState';
import { UnstyledLink, linkFrom } from '../UnstyledLink';
import { filterByVisible } from '../../helpers/array';

import styles from './Page.module.scss';

const Breadcrumb = ({ content, ...rest }) => (
  <UnstyledLink {...rest}>
    <ChevronLeft size={27} />
    {content}
  </UnstyledLink>
);

class Page extends Component {
  static displayName = 'Page';

  static defaultProps = {
    empty: {}
  }

  static propTypes = {
    /**
     * The Page display title
     */
    title: PropTypes.node,

    /**
     * Subtitle that appears to the right of the title
     */
    subtitle: PropTypes.node,

    /**
      * Main cta. Most button props will work in here.
      * e.g. { content: 'button label', onClick: callback() }
      */
    primaryAction: PropTypes.shape({
      content: PropTypes.node.isRequired
    }),

    /**
     * Alternative to primaryAction, accepts React nodes
     */
    primaryArea: PropTypes.node,

    /**
      * Actions that appear below title
      * e.g. { content: 'button label', onClick: callback() }
      */
    secondaryActions: PropTypes.arrayOf(PropTypes.shape({
      content: PropTypes.node.isRequired
    })),

    /**
      * The back link action
      */
    breadcrumbAction: PropTypes.shape({
      content: PropTypes.node.isRequired
    }),

    /**
      * Optional empty state object
      */
    empty: PropTypes.shape({
      show: PropTypes.bool,
      content: PropTypes.node
    }),

    /**
      * Page Children
      */
    children: PropTypes.node
  };

  render() {
    const {
      title,
      subtitle,
      primaryAction,
      primaryArea,
      secondaryActions,
      breadcrumbAction,
      empty,
      children
    } = this.props;

    const { show, content, ...emptyOptions } = empty;
    let subtitleMarkup = null;
    let primaryActionMarkup = null;

    if (show) {
      return (
        <EmptyState
          title={title}
          primaryAction={primaryAction}
          {...emptyOptions}>
          {content}
        </EmptyState>
      );
    }

    if (primaryAction) {
      primaryActionMarkup = buttonFrom(primaryAction, { size: 'large', ...(!primaryAction.color ? { color: 'blue' } : {}) });
    }

    if (primaryArea) {
      primaryActionMarkup = primaryArea;
    }

    const secondaryActionsMarkup = secondaryActions
      ? filterByVisible(secondaryActions).map((action, i) => linkFrom({ ...action, className: styles.SecondaryAction }, i))
      : null;

    const breadcrumbMarkup = breadcrumbAction
      ? <div><Breadcrumb {...breadcrumbAction} className={styles.Breadcrumb}/></div>
      : null;

    const titleMarkup = title ? <h1 className={styles.Title}>{title}</h1> : null;

    if (subtitle) {
      subtitleMarkup = typeof subtitle === 'string' ? <h2 className={styles.Subtitle}>{subtitle}</h2> : <div className={styles.SubtitleNode}>{subtitle}</div>;
    }

    return (
      <div>
        <div className={styles.Page}>
          {breadcrumbMarkup}
          <div className={styles.MainContent}>
            {titleMarkup}
            {subtitleMarkup}
            <div className={styles.PrimaryAction}>
              {primaryActionMarkup}
            </div>
          </div>
          <div className={styles.SecondaryActions}>
            {secondaryActionsMarkup}
          </div>
        </div>
        {children}
      </div>
    );
  }
}

export default Page;
