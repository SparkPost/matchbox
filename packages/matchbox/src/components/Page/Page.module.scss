@import '../../styles/config';

.Page {
  margin-top: spacing(larger);
  margin-bottom: spacing(larger);
}

.MainContent {
  display: flex;
  align-items: center;
}

.Title {
  flex: 0 0 0;
  margin-bottom: spacing(small);
  margin-top: spacing();
  padding: spacing(small) 0;
  @include text-style-display-heading;
  white-space: nowrap;
}

.Subtitle, .SubtitleNode {
  position: relative;
  flex: 1 0 0;
  margin-top: spacing();
  margin-left: rem(14);
  margin-bottom: 0;
  padding: spacing(small) 0 spacing(small) rem(14);

  &:before {
    display: inline-block;
    content: "";
    position: absolute;
    top: 50%;
    transform: translate(0, -50%);
    left: 0;
    width: 1px;
    height: rem(16);
    background: color(gray, 5);
  }
}

.Subtitle {
  font-size: font-size(600);
  font-weight: 500;
  color: color(gray, 1);
}

.PrimaryAction {
  flex: 1 0 0;
  margin-top: spacing();
  display: flex;
  justify-content: flex-end;
}

.SecondaryActions {
  margin-top: rem(-6);
}

.SecondaryAction, .SecondaryAction:visited {
  display: inline-block;
  padding: 0;
  margin-right: rem(24);

  color: color(gray, 3);
  font-size: font-size(600);
  text-decoration: none;
  transition: 0.15s;

  &:hover {
    color: color(gray, 1);
  }
}

.Breadcrumb, .Breadcrumb:visited {
  margin-left: rem(-3);
  margin-bottom: 0;
  text-decoration: none;
  color: color(blue);
  outline: none;

  *:first-child {
    margin-top: -2px;
  }

  &:hover, &:active, &:focus {
    color: color(blue, dark);
  }
}
