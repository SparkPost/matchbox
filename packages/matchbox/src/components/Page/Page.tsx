import React from 'react';
import styled from 'styled-components';
import { ChevronLeft, MoreHoriz } from '@sparkpost/matchbox-icons';
import { ActionList } from '../ActionList';
import { Box } from '../Box';
import { Button, ButtonProps } from '../Button';
import { EmptyState } from '../EmptyState';
import { Popover } from '../Popover';
import { UnstyledLink, UnstyledLinkProps } from '../UnstyledLink';
import { ScreenReaderOnly } from '../ScreenReaderOnly';
import { filterByVisible } from '../../helpers/array';

type ButtonActionProps = {
  content?: React.ReactNode;
  visible?: boolean;
};

type PageAction = ButtonProps & ButtonActionProps;

const StyledLink = styled(UnstyledLink)`
  text-decoration: none;
`;

type PageBreadcrumbProps = UnstyledLinkProps & {
  content?: React.ReactNode;
};

function Breadcrumb({ content, ...rest }: PageBreadcrumbProps): JSX.Element {
  return (
    <StyledLink {...rest} fontSize="200" lineHeight="200" fontWeight="medium">
      <Box as="span" display="inline-flex" alignItems="center">
        <ChevronLeft size="20" />
        {content}
      </Box>
    </StyledLink>
  );
}

type PageSubtitleProps = {
  subtitle?: React.ReactNode;
};

function Subtitle({ subtitle }: PageSubtitleProps): JSX.Element {
  if (!subtitle) {
    return null;
  }

  if (typeof subtitle === 'string') {
    return (
      <Box
        as="h2"
        fontSize={['300', null, null, '500']}
        lineHeight={['300', null, null, '500']}
        fontWeight="normal"
      >
        {subtitle}
      </Box>
    );
  }

  return <div>{subtitle}</div>;
}

type PageSecondaryActionsProps = {
  hasPrimaryAction?: boolean;
  actions?: PageAction[];
};

function SecondaryActions({ actions, hasPrimaryAction }: PageSecondaryActionsProps): JSX.Element {
  const visibleActions = React.useMemo(() => filterByVisible(actions), [actions]);
  const [isOpen, setIsOpen] = React.useState(false);

  if (!visibleActions.length) {
    return null;
  }

  if (visibleActions.length === 1) {
    const action = visibleActions[0];

    return (
      <Button
        {...action}
        variant="outline"
        color="blue"
        mr={hasPrimaryAction ? ['0', null, '300'] : ' 0'}
        ml={hasPrimaryAction ? ['300', null, '0'] : '0'}
      >
        {action.content}
      </Button>
    );
  }

  return (
    <Box
      position="relative"
      mr={hasPrimaryAction ? ['0', null, '300'] : ' 0'}
      ml={hasPrimaryAction ? ['300', null, '0'] : '0'}
    >
      <Popover
        id="page-secondary-actions"
        position={['bottomRight', null, 'bottomLeft']}
        onClose={() => setIsOpen(false)}
        open={isOpen}
        trigger={
          <Button
            aria-controls="page-secondary-actions"
            aria-expanded={isOpen}
            color="blue"
            onClick={() => setIsOpen(!isOpen)}
            variant="outline"
            p="0"
            width="2.5rem" // Forces a square
          >
            <MoreHoriz />
            <ScreenReaderOnly>More Options</ScreenReaderOnly>
          </Button>
        }
      >
        <ActionList>
          {visibleActions.map((action, i) => (
            <ActionList.Action key={i} {...action} />
          ))}
        </ActionList>
      </Popover>
    </Box>
  );
}

type PagePrimaryActionProps = {
  area?: React.ReactNode;
  action?: PageAction;
};

function PrimaryAction({ area, action }: PagePrimaryActionProps): JSX.Element {
  if (!action && !area) {
    return null;
  }

  if (area) {
    return <>{area}</>;
  }

  return (
    <Button color="blue" variant="filled" {...action}>
      {action.content}
    </Button>
  );
}

export type PageProps = {
  /**
   * The Page display title
   */
  title?: React.ReactNode;
  /**
   * Subtitle that appears next to the title
   */
  subtitle?: React.ReactNode;
  /**
   * Main cta. Most button props will work in here.
   * e.g. { content: 'button label', onClick: callback() }
   */
  primaryAction?: PageAction;
  /**
   * Alternative to primaryAction
   */
  primaryArea?: React.ReactNode;
  /**
   * Actions that appear in an Actionlist
   * e.g. { content: 'button label', onClick: callback() }
   */
  secondaryActions?: PageAction[];
  /**
   * The back link action
   */
  breadcrumbAction?: PageBreadcrumbProps;
  /**
   * Optional empty state object
   * @deprecated Use EmptyState instead
   */
  empty?: {
    show?: boolean;
    content?: React.ReactNode;
    [k: string]: any;
  };
  /**
   * Page Children
   */
  children?: React.ReactNode;
};

function Page(props: PageProps): JSX.Element {
  const {
    title,
    subtitle,
    primaryAction,
    primaryArea,
    secondaryActions,
    breadcrumbAction,
    empty = {},
    children,
  } = props;
  const { show, content, ...emptyProps } = empty;

  if (show) {
    return (
      // @ts-ignore
      <EmptyState.LEGACY title={title} primaryAction={primaryAction} {...emptyProps}>
        {content}
        {/* @ts-ignore */}
      </EmptyState.LEGACY>
    );
  }

  return (
    <div>
      <Box mt={['500', null, '700']} mb={['300', null, '450']}>
        {breadcrumbAction && (
          <Box mb="600">
            <Breadcrumb {...breadcrumbAction} />
          </Box>
        )}
        <Box display={[null, 'block', 'flex']} alignItems="center">
          <Box flex="1">
            {title && (
              <Box
                as="h1"
                fontWeight="semibold"
                fontSize={['500', null, '600']}
                lineHeight={['500', null, '600']}
              >
                {title}
              </Box>
            )}
            <Subtitle subtitle={subtitle} />
          </Box>
          <Box
            flex="0"
            display="flex"
            flexDirection={['row-reverse', null, 'row']}
            justifyContent="flex-end"
            mt={['400', null, '0']}
            mb={['400', null, '0']}
          >
            <SecondaryActions
              actions={secondaryActions}
              hasPrimaryAction={!!primaryAction || !!primaryArea}
            />
            <PrimaryAction area={primaryArea} action={primaryAction} />
          </Box>
        </Box>
      </Box>
      {children}
    </div>
  );
}

Page.displayName = 'Page';
export default Page;
