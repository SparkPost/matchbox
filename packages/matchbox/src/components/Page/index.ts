export { default as Page } from './Page';
export * from './Page';
