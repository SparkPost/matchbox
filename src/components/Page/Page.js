import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { buttonFrom } from '../Button';
import ChevronLeft from '../../icons/ChevronLeft';
import { EmptyState } from '../EmptyState';
import { UnstyledLink, linkFrom } from '../UnstyledLink';

import styles from './Page.module.scss';

const primaryOverrides = {
  primary: true,
  size: 'large'
};

const secondaryOverrides = {
  className: styles.SecondaryAction
};

const Breadcrumb = ({ content, ...rest }) => (
  <UnstyledLink {...rest}>
    <ChevronLeft size={27} />
    { content }
  </UnstyledLink>
);

class Page extends Component {
  static displayName = 'Page';

  static defaultProps = {
    empty: {}
  }

  static propTypes = {
    /**
     * The Page display title
     */
    title: PropTypes.string,
    /**
      * Main cta. Most button props will work in here.
      * e.g. { content: 'button label', onClick: callback() }
      */
    primaryAction: PropTypes.shape({
      content: PropTypes.string.isRequired
    }),
    /**
      * Actions that appear below title
      * e.g. { content: 'button label', onClick: callback() }
      */
    secondaryActions: PropTypes.arrayOf(PropTypes.shape({
      content: PropTypes.string.isRequired
    })),

    /**
      * The back link action
      */
    breadcrumbAction: PropTypes.shape({
      content: PropTypes.string.isRequired
    }),

    /**
      * Optional empty state object
      */
    empty: PropTypes.shape({
      show: PropTypes.bool,
      content: PropTypes.node
    }),

    /**
      * Page Children
      */
    children: PropTypes.node
  };

  render() {
    const {
      title,
      primaryAction,
      secondaryActions,
      breadcrumbAction,
      empty,
      children
    } = this.props;

    const { show, content, ...emptyOptions } = empty;

    if (show) {
      return (
        <EmptyState
          title={title}
          primaryAction={primaryAction}
          {...emptyOptions}>
            { content }
          </EmptyState>
      );
    }

    const primaryActionMarkup = primaryAction
      ? buttonFrom(primaryAction, primaryOverrides)
      : null;

    const secondaryActionsMarkup = secondaryActions
      ? secondaryActions.map((action, i) => linkFrom(
        { ...action, ...secondaryOverrides }, i
      ))
      : null;

    const breadcrumbMarkup = breadcrumbAction
      ? <Breadcrumb {...breadcrumbAction} className={styles.Breadcrumb}/>
      : null;

    const titleMarkup = title ? <h1 className={styles.Title}>{ title }</h1> : null;

    return (
      <div>
        <div className={styles.Page}>
          <div className={styles.Breadcrumb}>
            { breadcrumbMarkup }
          </div>
          <div className={styles.MainContent}>
            { titleMarkup }
            <div className={styles.PrimaryAction}>
              { primaryActionMarkup }
            </div>
          </div>
          <div className={styles.SecondaryActions}>
            { secondaryActionsMarkup }
          </div>
        </div>
        { children }
      </div>
    );
  }
}

export default Page;
