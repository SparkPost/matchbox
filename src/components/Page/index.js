export { default as Page } from './Page';
