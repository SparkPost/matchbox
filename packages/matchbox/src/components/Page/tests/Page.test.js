import React from 'react';
import { shallow } from 'enzyme';

import Page from '../Page';

describe('Page', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      title: 'Page Title',
      primaryAction: { content: 'Primary' }
    };

    wrapper = shallow(<Page {...props}><h1>Test Example</h1></Page>);
  });

  it('renders page with children', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders page with secondary actions', () => {
    wrapper.setProps({ secondaryActions: [{ content: 'secondary 1' }, { content: 'secondary 2' }]});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders page with breadcrumb', () => {
    wrapper.setProps({ breadcrumbAction: { content: 'Step 1' }});
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('Breadcrumb').dive()).toMatchSnapshot();
  });

  it('renders page with empty state', () => {
    wrapper.setProps({ empty: { show: true, content: 'Empty page' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('renders w/o title', () => {
    wrapper.setProps({ title: null });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders title as a node', () => {
    wrapper.setProps({ title: <div>test</div> });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders subtitle string', () => {
    wrapper.setProps({ subtitle: 'sub' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders subtitle node', () => {
    wrapper.setProps({ subtitle: <div>sub</div> });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders primary area node', () => {
    wrapper.setProps({ primaryArea: <div>primary</div> });
    expect(wrapper).toMatchSnapshot();
  });

});
