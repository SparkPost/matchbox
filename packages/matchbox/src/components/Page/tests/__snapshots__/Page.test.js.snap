// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders page with breadcrumb 1`] = `
<div>
  <div
    className="Page"
  >
    <div>
      <Breadcrumb
        className="Breadcrumb"
        content="Step 1"
      />
    </div>
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders page with breadcrumb 2`] = `
<UnstyledLink
  className="Breadcrumb"
>
  <ChevronLeft
    size={27}
  />
  Step 1
</UnstyledLink>
`;

exports[`Page renders page with children 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders page with empty state 1`] = `
<EmptyState
  primaryAction={
    Object {
      "content": "Primary",
    }
  }
  title="Page Title"
>
  Empty page
</EmptyState>
`;

exports[`Page renders page with secondary actions 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    >
      <UnstyledLink
        className="SecondaryAction"
        key="0"
      >
        secondary 1
      </UnstyledLink>
      <UnstyledLink
        className="SecondaryAction"
        key="1"
      >
        secondary 2
      </UnstyledLink>
    </div>
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders primary area node 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <div
        className="PrimaryAction"
      >
        <div>
          primary
        </div>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders subtitle node 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <div
        className="SubtitleNode"
      >
        <div>
          sub
        </div>
      </div>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders subtitle string 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        Page Title
      </h1>
      <h2
        className="Subtitle"
      >
        sub
      </h2>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders title as a node 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <h1
        className="Title"
      >
        <div>
          test
        </div>
      </h1>
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;

exports[`Page renders w/o title 1`] = `
<div>
  <div
    className="Page"
  >
    <div
      className="MainContent"
    >
      <div
        className="PrimaryAction"
      >
        <Button
          color="orange"
          size="large"
        >
          Primary
        </Button>
      </div>
    </div>
    <div
      className="SecondaryActions"
    />
  </div>
  <h1>
    Test Example
  </h1>
</div>
`;
